src/imports/resolveImports.ts

```diff
@@ -45,6 +46,15 @@ export function resolveImports({
                 continue;
             }
 
+            // Check for duplicates within the current file
+            if (resolvedPathsInCurrentFile.has(resolved.path)) {
+                throwCompilationError(
+                    `Duplicate import of '${resolved.path}' in file '${sourceFrom.path}'`,
```

Comment:
Hey thanks for the fix! Here the error message would be a bit too long and contain the full path, let's make it shorter